Tests/ScipioKitTests/Resources/Fixtures/TestingPackage/Package.swift

```diff
@@ -10,20 +10,38 @@ let package = Package(
         // Products define the executables and libraries a package produces, and make them visible to other packages.
         .library(
             name: "TestingPackage",
-            targets: ["TestingPackage"]),
+            targets: ["MyTarget"]
+        ),
+        .plugin(
+            name: "MyPlugin",
+            targets: ["MyPlugin"]
+        )
     ],
     dependencies: [
         .package(url: "https://github.com/apple/swift-log.git", .upToNextMinor(from: "1.4.4")),
     ],
     targets: [
-        // Targets are the basic building blocks of a package. A target can define a module or a test suite.
-        // Targets can depend on other targets in this package, and on products in packages this package depends on.
+        // Directly exported targets
         .target(
-            name: "TestingPackage",
+            name: "MyTarget",
             dependencies: [
                 .product(name: "Logging", package: "swift-log"),
             ]
         ),
+        .plugin(
+            name: "MyPlugin",
+            capability: .command(
+                intent: .custom(
+                    verb: "my-plugin-verb",
+                    description: "my-plugin-description")),
+            dependencies: ["ExecutableTarget"]
+        ),
```

Comment:
In my understood, SwiftPM doesn't support Plugin and Macros for Xcode build mode.
https://github.com/swiftlang/swift-package-manager/blob/release/6.0/Sources/XCBuildSupport/PIFBuilder.swift#L406-L411

I wonder it works well

Reply:
Certainly, plugins and macros are not meant to be built. However, we're talking here about the resolving phase where plugins may depend on executable targets, which in turn may depend on regular targets already intended to be built.

Correct me if I'm wrong:
There are two phases:
1. Resolve phase: all types of targets are important since any target may have a dependency on a regular target (which we are trying to identify while resolving).
2. Build phase: before starting, all irrelevant types of targets (everything except `.library` and `.binary`) [are filtered out anyway](https://github.com/giginet/Scipio/pull/132).

Is that right?

Reply:
You can see an example in the [swift-argument-parser](https://github.com/apple/swift-argument-parser/blob/main/Package.swift):

`GenerateManual (plugin)` -> `generate-manual(executableTarget)` -> `ArgumentParser(regularTarrget)` and `ArgumentParserToolInfo(regularTarget)`

Reply:
I see. You meant you want to make sure that the products are resolved in the correct order, regardless of whether it is built or not.

Your change seems to make sense.